### DISCUSSION TITLE
How to Define Number of Instances/Replicas for a Single Application?

### DISCUSSION THREAD

**Discussion by angelocala94** [coollabsio/coolify#2454] [Q&A]
Hi,

I am using Coolify to deploy a worker application based on BullMQ to handle background jobs. 
I need to run multiple instances of this worker in parallel to increase job processing capacity.

Is it possible to define the number of instances or replicas for a single application in Coolify? If so, could you please provide guidance on how to achieve this?

Thanks!

**Comment by CodingMeSwiftly** [comment]
Could you explain how you deployed a background worker? Coolify demands a port for application resources for me..

Regarding your issue, two solutions come to mind:
- Clone the resource `n` times
- Spawn `n` processes in a single resource with a [cluster util](https://www.npmjs.com/package/throng)

**Reply by angelocala94** [reply]
I used Docker Compose to define the infrastructure. This way, Nixpacks isn't used and Coolify doesn't assume your app needs to be exposed on the web.

To solve the multiple workers issue, I just defined the worker service multiple times in the Docker Compose file
